projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/usage-bar/index.tsx

```diff
@@ -27,9 +34,60 @@ const UsageBar: React.FC< UsageBarProps > = ( { usage } ) => {
 
 	return (
 		<div className="ai-assistant-usage-bar-wrapper">
-			<div className="ai-assistant-usage-bar-usage" style={ style }></div>
+			<div
+				className={ classNames( 'ai-assistant-usage-bar-usage', {
+					'is-limit-reached': limitReached,
+				} ) }
+				style={ style }
+			></div>
 		</div>
 	);
 };
 
-export default UsageBar;
+function UsageControl( {
+	isOverLimit,
+	hasFeature,
+	requestsCount,
+	requestsLimit,
+}: Pick< AIFeatureProps, 'isOverLimit' | 'hasFeature' | 'requestsCount' | 'requestsLimit' > ) {
+	let help = __( 'Unlimited requests for your site', 'jetpack' );
+
+	if ( ! hasFeature ) {
+		// translators: %1$d: number of requests allowed in the free plan
+		help = sprintf( __( '%1$d free requests for your site', 'jetpack' ), requestsLimit );
+	}
+
+	const limitReached = isOverLimit && ! hasFeature;
+	if ( limitReached ) {
+		help = __( 'You have reached your plan requests limit.', 'jetpack' );
+	}
+
+	// build messages
+	const freeUsageMessage = sprintf(
+		// translators: %1$d: current request counter; %2$d: request allowance;
+		__( '%1$d / %2$d free requests.', 'jetpack' ),
+		requestsCount,
+		requestsLimit
+	);
+	const unlimitedPlanUsageMessage = sprintf(
+		// translators: placeholder is the current request counter;
+		__( '%d / ∞ requests.', 'jetpack' ),
+		requestsCount
+	);
+
+	/*
+	 * Calculate usage. When hasFeature is true, the user has the paid plan,
+	 * that grants unlimited requests for now. To show something meaningful in
+	 * the usage bar, we use a very low usage value.
+	 */
```

Comment:
Since the usage bar is removed when `hasFeature` is true, this comment can be removed now